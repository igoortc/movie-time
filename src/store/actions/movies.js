import { FETCH_MOVIES, GET_MOVIE } from './actionTypes';

export const fetchMovies = keyword => ({
  type: FETCH_MOVIES,
  keyword,
});

export const getMovie = id => ({
  type: GET_MOVIE,
  id,
});
