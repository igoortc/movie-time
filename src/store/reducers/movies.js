import { FETCH_MOVIES_SUCCESS, FETCH_MOVIES_FAIL, GET_MOVIE_SUCCESS, GET_MOVIE_FAIL } from '../actions/actionTypes';

const initialState = {
  movies: [],
  loaded: false,
  error: false,
  movie: {},
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_MOVIES_SUCCESS:
      return {
        ...state,
        loaded: true,
        movies: action.movies,
        error: false,
      };
    case FETCH_MOVIES_FAIL:
      return {
        ...state,
        loaded: false,
        error: true,
      };
    case GET_MOVIE_SUCCESS:
      return {
        ...state,
        loaded: true,
        movie: action.movies,
        error: false,
      };
    case GET_MOVIE_FAIL:
      return {
        ...state,
        loaded: false,
        error: true,
      };
    default:
      return state;
  }
};
