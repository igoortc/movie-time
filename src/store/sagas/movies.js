import { put } from 'redux-saga/effects';
import actions from '../actions/actionTypes';
import api from '../../utils/api';

export function* fetchMoviesSaga(action) {
  try {
    const response = yield api.fetchMovies(action.keyword);
    const movies = yield response.data.results;
    movies.length > 0
      ? yield put({ type: actions.FETCH_MOVIES_SUCCESS, movies: movies })
      : yield put({ type: actions.FETCH_MOVIES_FAIL });
  } catch (error) {
    yield put({ type: actions.FETCH_MOVIES_FAIL, message: error.message });
  }
}

export function* getMovieSaga(action) {
  try {
    const response = yield api.getMovie(action.id);
    const movies = yield response.data;
    yield put({ type: actions.GET_MOVIE_SUCCESS, movies: movies });
  } catch (error) {
    yield put({ type: actions.GET_MOVIE_FAIL, message: error.message });
  }
}
