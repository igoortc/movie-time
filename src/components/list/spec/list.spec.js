import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import movies from '../../../utils/movies';
import List from '..';

describe('List', () => {
  it('should render List correctly', () => {
    const component = shallow(
      <BrowserRouter>
        <List movies={movies} />
      </BrowserRouter>
    );

    expect(component).toMatchSnapshot();
  });
});
