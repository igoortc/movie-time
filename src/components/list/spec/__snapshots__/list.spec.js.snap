// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render List correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(List)
    movies={
      Array [
        Object {
          "adult": false,
          "backdrop_path": "/hbn46fQaRmlpBuUrEiFqv0GDL6Y.jpg",
          "genre_ids": Array [
            878,
            28,
            12,
          ],
          "id": 24428,
          "original_language": "en",
          "original_title": "The Avengers",
          "overview": "When an unexpected enemy emerges and threatens global safety and security, Nick Fury, director of the international peacekeeping agency known as S.H.I.E.L.D., finds himself in need of a team to pull the world back from the brink of disaster. Spanning the globe, a daring recruitment effort begins!",
          "popularity": 47.408,
          "poster_path": "/cezWGskPY5x7GaglTTRN4Fugfb8.jpg",
          "release_date": "2012-04-25",
          "title": "The Avengers",
          "video": false,
          "vote_average": 7.6,
          "vote_count": 17839,
        },
        Object {
          "adult": false,
          "backdrop_path": "/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg",
          "genre_ids": Array [
            12,
            28,
            14,
          ],
          "id": 299536,
          "original_language": "en",
          "original_title": "Avengers: Infinity War",
          "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.",
          "popularity": 109.856,
          "poster_path": "/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
          "release_date": "2018-04-25",
          "title": "Avengers: Infinity War",
          "video": false,
          "vote_average": 8.3,
          "vote_count": 11138,
        },
        Object {
          "adult": false,
          "backdrop_path": "/aw8niyYZNTonD1ep5vMZm9LZ26Z.jpg",
          "genre_ids": Array [
            12,
            878,
            28,
          ],
          "id": 299534,
          "original_language": "en",
          "original_title": "Avengers: Endgame",
          "overview": "A culmination of 22 interconnected films and the finale of this epic journey across the length and breadth of the Marvel Cinematic Universe.  Our beloved heroes will truly understand how fragile this reality is and the sacrifices that must be made to uphold it.",
          "popularity": 33.795,
          "poster_path": "/bJLYrLIHT1r7cikhWGbpZkxlUpA.jpg",
          "release_date": "2019-04-24",
          "title": "Avengers: Endgame",
          "video": false,
          "vote_average": 0,
          "vote_count": 0,
        },
        Object {
          "adult": false,
          "backdrop_path": "/rFtsE7Lhlc2jRWF7SRAU0fvrveQ.jpg",
          "genre_ids": Array [
            28,
            12,
            878,
          ],
          "id": 99861,
          "original_language": "en",
          "original_title": "Avengers: Age of Ultron",
          "overview": "When Tony Stark tries to jumpstart a dormant peacekeeping program, things go awry and Earthâ€™s Mightiest Heroes are put to the ultimate test as the fate of the planet hangs in the balance. As the villainous Ultron emerges, it is up to The Avengers to stop him from enacting his terrible plans, and soon uneasy alliances and unexpected action pave the way for an epic and unique global adventure.",
          "popularity": 33.106,
          "poster_path": "/t90Y3G8UGQp0f0DrP60wRu9gfrH.jpg",
          "release_date": "2015-04-22",
          "title": "Avengers: Age of Ultron",
          "video": false,
          "vote_average": 7.3,
          "vote_count": 11948,
        },
        Object {
          "adult": false,
          "backdrop_path": "/8YW4rwWQgC2JRlBcpStMNUko13k.jpg",
          "genre_ids": Array [
            53,
            878,
          ],
          "id": 9320,
          "original_language": "en",
          "original_title": "The Avengers",
          "overview": "British Ministry agent John Steed, under direction from \\"Mother\\", investigates a diabolical plot by arch-villain Sir August de Wynter to rule the world with his weather control machine. Steed investigates the beautiful Doctor Mrs. Emma Peel, the only suspect, but simultaneously falls for her and joins forces with her to combat Sir August.",
          "popularity": 7.245,
          "poster_path": "/zxjJ2IxhilZLzl8eXjj7S0apZ7k.jpg",
          "release_date": "1998-08-13",
          "title": "The Avengers",
          "video": false,
          "vote_average": 4.3,
          "vote_count": 308,
        },
        Object {
          "adult": false,
          "backdrop_path": "/cXHiXZWTPMClrJazhi7ZaB7Y3iK.jpg",
          "genre_ids": Array [
            28,
            16,
            12,
            878,
          ],
          "id": 14609,
          "original_language": "en",
          "original_title": "Ultimate Avengers",
          "overview": "When a nuclear missile was fired at Washington in 1945, Captain America managed to detonate it in the upper atmosphere. But then he fell miles into the icy depths of the North Atlantic, where he remained lost for over sixty years. But now, with the world facing the very same evil, Captain America must rise again as our last hope for survival.",
          "popularity": 7.206,
          "poster_path": "/pmNYggLktcWHzQeCru5ywL4OiRH.jpg",
          "release_date": "2006-02-21",
          "title": "Ultimate Avengers",
          "video": false,
          "vote_average": 6.5,
          "vote_count": 132,
        },
        Object {
          "adult": false,
          "backdrop_path": "/85NqI4WuCim6dZexmTPUAi13Af0.jpg",
          "genre_ids": Array [
            12,
            16,
            28,
            878,
          ],
          "id": 14611,
          "original_language": "en",
          "original_title": "Ultimate Avengers 2",
          "overview": "Mysterious Wakanda lies in the darkest heart of Africa, unknown to most of the world. An isolated land hidden behind closed borders, fiercely protected by its young king - the Black Panther. But when brutal alien invaders attack, the threat leaves the Black Panther with no option but to go against the sacred decrees of his people and ask for help from outsiders.",
          "popularity": 7.586,
          "poster_path": "/u7vvexSU81Qk20yU7Vog23Ogob.jpg",
          "release_date": "2006-08-08",
          "title": "Ultimate Avengers 2",
          "video": false,
          "vote_average": 6.3,
          "vote_count": 112,
        },
      ]
    }
  />
</Router>
`;
